src/ pages/Main/Main.js

```diff
@@ -1,8 +1,63 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './Main.scss';
+import Collection from './Collection/Collection';
+import Banner from './Banner/Banner';
+import Nav from './Nav/Nav';
+import LoadMoreProducts from './LoadMoreProducts/LoadMoreProducts';
+import Skeleton from './Skeleton/Skeleton';
+import FilterProduct from './FilterProduct/FilterProduct';
 
 function Main() {
-  return <div>main</div>;
+  const [productsList, setProductsList] = useState([]);
+  const [cartList, setCartList] = useState([]);
+  const [page, setPage] = useState(1);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchProductsData = async pageNum => {
+      if (pageNum < 5) {
+        const response = await fetch(
+          `/data/main/MainProductList${pageNum}.json`
+        );
+        const data = await response.json();
+        setProductsList(productsList => [...productsList, ...data]);
+      }
+    };
+    const loadFirstTime = async () => {
+      await setIsLoading(true);
+      await fetchProductsData(page);
+      await setIsLoading(false);
+    };
+    loadFirstTime();
+  }, [page]);
+
+  function addCart(product) {
+    setCartList([...cartList, product.id]);
+  }
+
+  return (
+    <>
+      <div className="main">
+        <Nav cartCount={cartList.length} />
+        {/* 테스트를 위해 임시로 만든 Nav 컴포넌트 */}
+        {isLoading && <Skeleton />}
+        <Banner />
+        {productsList.map((products, idx) => (
+          <Collection
+            key={idx}
+            products={products}
+            addCart={addCart}
+            cartList={cartList}
+            showMore={true}
+          />
+        ))}
+      </div>
+      {page === 5 && (
+        <FilterProduct addCart={addCart} cartList={cartList} showMore={false} />
+      )}
+      {!isLoading && page < 5 ? <LoadMoreProducts setPage={setPage} /> : ''}
```

Comment:
```suggestion
      {!isLoading && page < 5 && <LoadMoreProducts setPage={setPage} />}
```
조건이 true 일 때만 렌더링하도록 단축 평가를 이용해줄 수 있습니다

Reply:
수정했습니다!